Updated comment on external vs public functions in Solidity

Akkii4 opened the pull request:
For Solidity versions after 0.8.0, there is no longer a gas cost difference between external and public function calls.

Akkii4 committed:
Updated comment on external vs public functions in Solidity

For Solidity versions after 0.8.0, there is no longer a gas cost difference between external and public function calls.

patrickd- commented on contracts/cheatsheet.sol:
```diff
@@ -428,7 +427,7 @@ contract Test is ITest, AdvTestBase, Storage, ownerNamespace.Owner, Ballot {
     /**
      * If you have a function that will be called very often, in order to safe gas..
-     *  - declare functions as external and make use of calldata location for parameters.
+     *  - declare functions as external/public and make use of calldata location for parameters.
```
Actually, not only external/public functions can make use of calldata location for parameters. Private/Internal functions can too (eg. if another externally called function passes its calldata to it).

maybe it would make sense to write it this way?

```diff
-     *  - declare functions as external and make use of calldata location for parameters.
+     *  - functions that will be called externally can make use of calldata location for parameters.
```

wdyt?

Akkii4 replied:
It's seems perfect

Akkii4 committed:
updated comment on use of calldata parameter

modified the comment based on the discussion with patrickd on the use of calldata param in order to save gas
